src/libraries/System.Formats.Tar/tests/TarFile/TarFile.ExtractToDirectory.File.Tests.cs

```diff
@@ -71,6 +71,49 @@ public void SetsLastModifiedTimeOnExtractedFiles()
             Assert.InRange(File.GetLastWriteTime(outFile).Ticks, dt.AddSeconds(-3).Ticks, dt.AddSeconds(3).Ticks); // include some slop for filesystem granularity
         }
 
+        [Fact]
+        public void SetsLastModifiedTimeOnExtractedDirectories()
+        {
+            using TempDirectory root = new TempDirectory();
+
+            DirectoryInfo fromDir = Directory.CreateDirectory(Path.Combine(root.Path, "fromdir"));
+            // Create a hierarcy of directories.
+            var directories = new DirectoryInfo[]
+            {
+                Directory.CreateDirectory(Path.Combine(fromDir.FullName, "dir")),                      // 'fromdir/dir'
+                Directory.CreateDirectory(Path.Combine(fromDir.FullName, "dir", "child")),             // 'fromdir/dir/child'
+                Directory.CreateDirectory(Path.Combine(fromDir.FullName, "dir", "child", "subchild")), // 'fromdir/dir/child/subchild'
+                Directory.CreateDirectory(Path.Combine(fromDir.FullName, "dir2")),                     // 'fromdir/dir2'
+                Directory.CreateDirectory(Path.Combine(fromDir.FullName, "dir2", "child2")),           // 'fromdir/dir2/child'
+            };
+            var dt = new DateTime[directories.Length];
+            for (int i = directories.Length - 1; i >= 0; i--) // Reverse order to preserve parent timestamps.
+            {
+                // Add a file.
+                File.Create(Path.Combine(directories[i].FullName, "file")).Dispose();
+
+                // Set the directory timestamp.
+                dt[i] = new DateTime(2000 + i, 1, 2, 3, 4, 5, DateTimeKind.Local);
```

Comment:
ideally this test would set a unique time on all the files and all the directories in the tarball, and verify every one is as we expect.
as it is, the files are getting the current timestamp, and the directories are all sharing an old timestamp.

Reply:
There's a separate test for files (`SetsLastModifiedTimeOnExtractedFiles`). This test focuses on directories. Some files are added to ensure they don't mess up the directory timestamps.

Reply:
gotcha (although why not make it all one test?) but I do think the timestamps should all be unique.